lib/drives/models/drive.dart

```diff
@@ -57,4 +59,25 @@ class Drive extends Model {
   String toString() {
     return 'Drive{id: $id, from: $start at $startTime, to: $end at $endTime, by: $driverId}';
   }
+
+  static Future<Drive?> userAlreadyHasDrive(
+      DateTime start, DateTime end, int userId) async {
+    final List<dynamic> drivesList =
+        await supabaseClient.from('drives').select().eq('driver_id', userId);
+    final List<Drive> drives = drivesList
+        .map((drive) => Drive.fromJson(drive as Map<String, dynamic>))
+        .toList();
+    for (Drive drive in drives) {
+      if (drive.startTime.isBefore(start) && drive.endTime.isAfter(start)) {
```

Comment:
these three cases can be simplified into one:
`if(drive.startTime.isBefore(end) && drive.endTime.isAfter(start))`

Reply:
yes i you are right changed that.